pkg/observability/observability.go

```diff
@@ -0,0 +1,98 @@
+package observability
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"path"
+	"strings"
+	"text/template"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+)
+
+func CreatePrometheusConfigs(
```

Comment:
Is it the same:
```
func CreatePrometheusConfigs(
	ctx context.Context,
	cli client.Client,
	enabled bool,
	rootFS embed.FS,
	manifestPath string,
	owner metav1.Object,
	dscispec *dsciv1.DSCInitializationSpec) error {

	foundObj := &unstructured.Unstructured{}
	entries, err := rootFS.ReadDir(manifestPath)
	if err != nil {
		return fmt.Errorf("error reading dir %w", err)
	}
	for _, e := range entries {
		// Should not be a bit deal to generate it for Get() real error, should not happen in normal flow
		object, err := updateTemplate(e.Name(), rootFS, manifestPath, dscispec)
		if err != nil {
			return fmt.Errorf("failed inject template for PrometheusRules CR: %w", err)
		}

		resourceName := strings.Split(e.Name(), ".")[0]
		foundObj := setGVK(foundObj, resourceName)
		err = cli.Get(ctx, client.ObjectKey{Name: resourceName, Namespace: dscispec.Monitoring.Namespace}, foundObj)

		switch {
		// object exists
		case err == nil:
			// if enabled, nothing
			if !enabled {
				if err = cli.Delete(ctx, object); err != nil {
					return fmt.Errorf("error removing %s: %w", resourceName, err)
				}
			}
		case apierrs.IsNotFound(err):
			if enabled {
				if err = ctrl.SetControllerReference(owner, metav1.Object(object), cli.Scheme()); err != nil {
					return fmt.Errorf("error setting owner reference for %s: %w", resourceName, err)
				}
				if err = cli.Create(ctx, object); err != nil {
					return fmt.Errorf("error creating %s: %w", resourceName, err)
				}
			}
		default:
			return fmt.Errorf("failed fetching %v CR: %w", resourceName, err)
		}

	}
	return nil
}
```

Reply:
oh yes. this refractor is much better :). Will update the code accordingly. Thanks